src/common/src/range.rs

```diff
@@ -75,6 +75,12 @@ pub trait RangeBoundsExt<T: Idx>: RangeBounds<T> {
         }
     }
 
+    fn bounds(&self, start: T, end: T) -> Range<T> {
```

Comment:
This usage seems uncommon and the method name is confusing. Better to inline the logic instead of having a extra method call.